### DISCUSSION TITLE
Compile unicode file from git

### DISCUSSION THREAD

**Discussion by BallonLac** [codefori/vscode-ibmi#777] [Q&A] [answered]
Hi there!

First, I'm an absolute beginner with VS Code and git (only using SEU/PDM in green screen for now, with local files).
Apologies if my question is covered elsewhere or not relevant.
And, BTW, many thanks for all your good work, this project is awesome! :)

I'm running a V7R4 test partition, installed with 2928 (French) language.
QCCSID is 65535, so I changed my profile for CCSID=1174 (French Euro) and LANGID=FRA.
No language-specific variables in my Unix environnement (I'm using Bash).
I didn't change anything in VS Code settings (SQL is enabled).
No problem with compiling some dummy programs in my libraries (got them from your videos, thanks !).

As a beginner's training with using IFS in VS Code, I'm trying to compile something useful: https://github.com/worksofliam/migrate.
The `git clone` was OK, I just chmod "src" dir and files to 777.
Then, I installed gmake from make-gnu 4-2.2 (source @ibm).

In a PASE terminal, going to src dir and lauching `gmake` gives me:
> -bash-5.1$ gmake
> system "CRTRPGMOD MODULE(MIGRATE/migsrcpf) SRCSTMF('src/migsrcpf.rpgle') DBGVIEW(*NONE)"
> RNS9339: Impossible d'ouvrir le fichier src/migsrcpf.rpgle.
> RNS9309: La compilation a échoué. Le module MIGSRCPF n'a pas été créé dans la bibliothèque MIGRATE.
> gmake: *** [makefile:20: migsrcpf.rpgle] Error 255

(sorry for the french)

As all files are 777, I tried to run the exact same `CRTRPGMOD` command in a green screen, having my curdir changed to my src dir.
What I can see in my joblog, is a RNS9380 error (followed by the RNS9339):

>  ID message . . . . . . :   RNS9380                                             
>  Date d'envoi . . . . . :   07/24/22      Heure d'envoi  . . . . :   15:40:11   
>                                                                                 
>  Message . . . . :   Le CCSID de fichier source 1208 est un CCSID Unicode qui   
>    ne peut pas être utilisé avec TGTCCSID(*SRC).                                
>                                                                                 
>  Cause . . . . . :   Lorsque TGTCCSID(*SRC) est spécifié, le CCSID du fichier   
>    source principal ne peut pas être un CCSID Unicode.                          
>  Que faire . . . :   Indiquez *JOB ou une valeur EBCDIC valide pour le          
>    paramètre TGTCCSID.                                                          
>  Technical description . . . . . . . . :   TGTCCSID(*SRC) is only supported     
>    when the CCSID of the primary source file is an EBCDIC or ASCII CCSID.       
                                                                                 
So, if I understand this message the problem is not a lang problem, but with encoding, right?
Maybe I did something wrong, or my green screen result is irrelevant?
Any advice?

Thanks :)

**Comment by chrjorgensen** [comment, marked as answer]
Hi.

IBM i 7.4 PASE stores files in UTF-8 as default - so your RPG source code is stored in UTF-8.
Check this article on how to [Compile RPG from Unicode source](https://www.ibm.com/support/pages/node/1108071)

This is not an issue with Code for IBM i or VS Code - I suggest you read the info on [RPG Café](https://ibm.biz/rpg_cafe) for RPG news and issues. And connect to [Ryver](https://ibmioss.ryver.com/index.html#forums/1000141/chat), where there is a community for IBM i and open source with forums to ask questions like this. All the IBM i experts are there... 😉

**Comment by BallonLac** [comment]
Hi,

Thanks for your reply, it gives me a lot to read & learn :)
